exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/statistics/StatisticsMergeBatch.java

```diff
@@ -0,0 +1,532 @@
+          }
+          if (statMap.get(colName) != null) {
+            hllHolder = (ObjectHolder) statMap.get(colName);
+          } else {
+            hllHolder = new ObjectHolder();
+            hllHolder.obj = new HyperLogLog(context.getContextInformation().getHllAccuracy());
+            statMap.put(colName, hllHolder);
+          }
+          NullableVarBinaryVector hllVector = (NullableVarBinaryVector) mapElt;
+          try {
+            if (hllVector.getAccessor().isSet(0) == 1) {
+              ByteArrayInputStream bais = new ByteArrayInputStream(hllVector.getAccessor().getObject(0), 0,
+                  mapElt.getBufferSize());
+              HyperLogLog other = HyperLogLog.Builder.build(new DataInputStream(bais));
+              ((HyperLogLog) hllHolder.obj).addAll(other);
+            }
+          } catch (Exception ex) {
+            //TODO: Catch IOException/CardinalityMergeException
+            //TODO: logger
+            return IterOutcome.STOP;
+          }
+        }
+      }
+      // Add NDV value vector map using HLL map (since the NDV map is directly generated from HLL and not produced by the underlying
+      // Statistics Agg)
+      Map<String, ValueHolder> hllMap = aggregationMap.get("hll");
+      if (hllMap != null) {
+        aggregationMap.put("ndv", hllMap);
+      }
+    }
+    return IterOutcome.OK;
+  }
+
+  public VectorContainer getOutgoingContainer() {
+    return this.container;
+  }
+
+  @Override
+  public IterOutcome innerNext() {
+    IterOutcome outcome;
+    boolean didSomeWork = false;
+    if (finished) {
+      return IterOutcome.NONE;
+    }
+    try {
+      outer: while (true) {
+        outcome = next(incoming);
+        switch (outcome) {
+          case NONE:
+            break outer;
+          case OUT_OF_MEMORY:
+          case NOT_YET:
+          case STOP:
+            return outcome;
+          case OK_NEW_SCHEMA:
+            if (first) {
+              first =false;
+              if (!setupNewSchema()) {
+                outcome = IterOutcome.OK;
+              }
+              return outcome;
+            }
+            //fall through
+          case OK:
+            assert first == false : "First batch should be OK_NEW_SCHEMA";
+            IterOutcome out = doWork();
+            didSomeWork = true;
+            if (out != IterOutcome.OK) {
+              return out;
+            }
+            break;
+          default:
+            throw new UnsupportedOperationException("Unsupported upstream state " + outcome);
+        }
+      }
+    } catch (SchemaChangeException ex) {
+      kill(false);
+      logger.error("Failure during query", ex);
+      context.fail(ex);
+      return IterOutcome.STOP;
+    }
+
+    // We can only get here if upstream is NONE i.e. no more batches. If we did some work prior to
+    // exhausting all upstream, then return OK. Otherwise, return NONE.
+    if (didSomeWork) {
+      IterOutcome out = buildOutgoingRecordBatch();
```

Comment:
For the parallel plan, we would get a record batch from each of the minor fragments. Hence, the multiple batches.